frontend/app/work_packages/services/work-package-field-service.js

```diff
@@ -53,6 +53,9 @@ module.exports = function(
     // TODO: extract to strategy if new cases arise
     if (field === 'date') {
       // nope
+      if(workPackage.links.children) {
+        return false;
+      }
```

Comment:
I think you should be able to erase both [special cases](https://github.com/hirtie-maxim/openproject/blob/be7a16205319d070c2e27b64fa697f8888e2405a/frontend/app/work_packages/services/work-package-field-service.js#L54-L65) from this method.